### PR TITLE
Camrossi nae5.1 support

### LINKED ISSUE #15: Requirement Set API endpoint changed in NAE 5.1
As of NAE 5.1 this end point: 

`https://%(host)s:%(port)s/nae/api/v1/event-services/assured-networks/%(fabric_uuid)s/model/aci-policy/compliance-requirement/requirement_sets/%(obj_uuid)s`

Is now 

`https://%(host)s:%(port)s/nae/api/v1/event-services/assured-networks/%(fabric_uuid)s/model/aci-policy/compliance-requirement/requirement-sets/%(obj_uuid)s`

the requirement_sets became requirement-sets

### LINKED ISSUE #16: resp.read() should only be called once
In NAE 5.1 any call to resp.read() after the first one, will return an empty array. 
All the code should change to accomodate this. 

For example this code fails
```
self.result['Result'] = json.loads(resp.read())['value']['data']
return json.loads(resp.read())['value']['data']
```
Every instance should become
```
r = resp.read()
self.result['Result'] = json.loads(r)['value']['data']
return json.loads(r)['value']['data']
```

### LINKED ISSUE #17: Typo in def get_compliance_object
This is not correct and cause the code not to work. 

` elif self.params.get('selector') == 'requirement_sets':`

It should be `requirement_set `

### PR DESCRIPTION
**Pull request by camrossi**
Fixes #15 #16 #17

### CODE REVIEW

File: plugins/module_utils/nae.py [hunk 1]
```diff
@@ -1027,8 +1032,12 @@ def new_object_selector(self):
                     msg='Connection failed for %(url)s. %(msg)s' %
                     auth, **self.result)
         else:
+            if resp.headers['Content-Encoding'] == "gzip":
```
Comment (xinyuezhao) [inline plugins/module_utils/nae.py:1035]: it might be better to use resp.headers.get('Content-Encoding')

**Review by xinyuezhao — COMMENTED** [review]
review it together with PR #24

**Comment by lhercot** [comment]
Looks like this one is part of your second PR so should we just close this one?

**Comment by camrossi** [comment]
Yes @lhercot we can do it all in the other PR

**Comment by lhercot** [comment]
Closing as PR #24 already contains those changes.

**Pull request closed without merging**